mod_app/static/ckeditor/ckeditor/plugins/popout/plugin.js

```diff
@@ -0,0 +1,104 @@
+
+        var editorElement = editor.element.$;
+        var editorContainer = editorElement.closest(".c-2");
+        var labelElement =
+          editorContainer.previousElementSibling.querySelector("label"); // Find the label in the sibling 'c-1'
+        var labelText = labelElement
+          ? labelElement.innerText
+          : editor.name.split("id_").pop();
+
+        var instanceTitle = document.getElementById("id_title").value;
+        var path = window.location.pathname.split("mod_app/").pop();
+
+        // Remove the trailing "/change"
+        if (path.endsWith("/change")) {
+          path = path.slice(0, -7);
+        }
+        var windowTitle = instanceTitle || path;
+
+        // html structure for new window
+        newWindow.document.write(
+          `<html><head>              <title>MOD Editor || ${windowTitle} </title>
+</head><body>
+<h3 style="font-family:sans-serif">Editing ${labelText} for ${instanceTitle}</h3><textarea id="popout-editor"></textarea>
+<button id="save-button" style="${btnStyle}">Save</button><button id="save-close-button" style="${btnStyle}">Save & Close</button></body></html>`
+        );
+
+        newWindow.document.close();
+
+        // Load CKEditor script in the new window
+        var script = newWindow.document.createElement("script");
+        script.type = "application/javascript";
+        script.src = ckeditorBasePath + "ckeditor.js";
+
+        script.onload = function () {
+          newWindow.CKEDITOR.replace("popout-editor", {
+            ...originalConfig,
+            // pass existing editor content to the new editor
+            on: {
+              instanceReady: function () {
+                newWindow.CKEDITOR.instances["popout-editor"].setData(
+                  editor.getData()
+                );
+              },
+            },
+          });
+
+          // Sync data from the new window to the original editor
+          newWindow.CKEDITOR.instances["popout-editor"].on(
+            "change",
+            function () {
+              editor.setData(
+                newWindow.CKEDITOR.instances["popout-editor"].getData()
+              );
+            }
+          );
+
+          // add listener to save button
+          newWindow.document
+            .getElementById("save-close-button")
+            .addEventListener("click", function () {
+              // Transfer content from the new editor to the original editor
+              editor.setData(
+                newWindow.CKEDITOR.instances["popout-editor"].getData()
+              );
+
+              // Close the new window
+              newWindow.close();
+            });
+
+          newWindow.document
+            .getElementById("save-button")
+            .addEventListener("click", function () {
+              // Transfer content from the new editor to the original editor
+              editor.setData(
+                newWindow.CKEDITOR.instances["popout-editor"].getData()
+              );
+            });
```

Comment:
I don't understand the need for this if the content is already being synced on change (lines 62-69).

Reply:
ah whoops, I meant to delete the syncing; it only works if you're quite slow so opted for the save options